skimage/restoration/deconvolution.py

```diff
@@ -365,13 +365,22 @@ def richardson_lucy(image, psf, iterations=50, clip=True):
     ----------
     .. [1] http://en.wikipedia.org/wiki/Richardson%E2%80%93Lucy_deconvolution
     """
+    direct_time = lambda n, m, k, l: k*l * n*m
+    def fft_time(m, n, k, l):
+        return m*np.log(m) + n*np.log(n) + k*np.log(k) + l*np.log(l)
+
+    time_ratio = 71.468 * fft_time(*image.shape, *psf.shape)
```

Comment:
However, I would change all this quite dramatically, since it takes almost no effort to make this function work on nD images instead of just 2D:

```python
def direct_time(im, psf):
    return np.product(im.shape + psf.shape)
def fft_time(im, psf):
    return sum(k*np.log(k) for k in im.shape + psf.shape)
def time_ratio(im, psf):
    # constant factor obtained by curve fitting, see
    # https://github.com/scikit-image/scikit-image/pull/1792
    return 71.468 * fft_time(im, psf) / direct_time(im, psf)
convolve_function = fftconvolve if time_ratio(im, psf) < 1 else convolve
```
By using the `convolve` function and the more general function definitions, we get nD functionality essentially for free!

The only downside is that the constant factors might change now. =\